src/public/media.ts

```diff
@@ -141,6 +142,47 @@ export namespace media {
         callback(invalidInput, null);
         return;
       }
+      // Call the new get media implementation via callbacks if the client version is greater than or equal to '2.0.0'
+      if (callGetMediaViaCallback('2.0.0')) {
+        this.getMediaViaCallback(callback);
+      } else {
+        this.getMediaViaHandler(callback);
+      }
+    }
+
+    private getMediaViaCallback(callback: (error: SdkError, blob: Blob) => void): void {
+      const helper: MediaHelper = {
+        mediaMimeType: this.mimeType,
+        assembleAttachment: [],
+      };
+      const params = [this.content];
+      const actionName = sendMessageRequestToParent('getMedia', params);
+      function handleGetMediaCallbackRequest(mediaResult: MediaResult): void {
+        if (callback) {
+          if (mediaResult && mediaResult.error) {
+            callback(mediaResult.error, null);
+          } else {
+            if (mediaResult && mediaResult.mediaChunk) {
+              // If the chunksequence number is less than equal to 0 implies EOF
+              // create file/blob when all chunks have arrived and we get 0/-1 as chunksequence number
+              if (mediaResult.mediaChunk.chunkSequence <= 0) {
+                const file = createFile(helper.assembleAttachment, helper.mediaMimeType);
+                callback(mediaResult.error, file);
+              } else {
+                // Keep pushing chunks into assemble attachment
+                const assemble: AssembleAttachment = decodeAttachment(mediaResult.mediaChunk, helper.mediaMimeType);
+                helper.assembleAttachment.push(assemble);
+              }
+            } else {
+              callback({ errorCode: ErrorCode.INTERNAL_ERROR, message: 'data receieved is null' }, null);
+            }
+          }
+        }
+      }
+      GlobalVars.callbacks[actionName] = handleGetMediaCallbackRequest;
+    }
+
+    private getMediaViaHandler(callback: (error: SdkError, blob: Blob) => void): void {
```

Comment:
Keeping the old code path around totally negates the purpose of this change; we are just adding more complexity and confusion to the code. Can we remove the old code path? If not, then what's the plan for cleaning this up?

Reply:
We will have to keep the old code for now to ensure backward compatibility. Ideally we can clean it up once we are sure every client's client version is greater than or equal to 2.0.0. But I don't know when that would be or how to track that.

Reply:
Create a work item to track telemetry and delete the old code once we determine that it is not being used anymore by anyone.

Reply:
also: call delete GlobalVars.handlers['getMedia' + actionName] instead of setting it to null.

Reply:
done. https://domoreexp.visualstudio.com/MSTeams/_workitems/edit/1376437